src/tequila/circuit/_gates_impl.py

```diff
@@ -376,11 +376,38 @@ def extract_targets(generator):
             targets += [k for k in ps.keys()]
         return tuple(set(targets))
 
-    def __init__(self, angle, generator, control=None, eigenvalues_magnitude=0.5, steps=1, assume_real=False):
-        super().__init__(eigenvalues_magnitude=eigenvalues_magnitude, assume_real=assume_real, name="GenRot", parameter=angle, target=self.extract_targets(generator), control=control)
+    def __init__(self, angle, generator, p0=None, control=None, eigenvalues_magnitude=0.5, steps=1, name="GenRot", assume_real=False):
```

Comment:
needs a small modification here:
init should take: `target=None`
afterwards:
```python
if target is None:
    target = self.extract_targets(generator)
```

and then pass `target` to the `super().__init__`

Reason:  The optimized compile routines for QubitExcitations need the target qubits in the correct order (which is lost if the target qubits are just extracted from the generator).

Reply:
What a coincidence, I just realized it the very moment your comment popped up in my code editor 😁 should be fixed in the last commit.